dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/SingularityIT.java

```diff
@@ -0,0 +1,140 @@
+package io.dockstore.client.cli;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+
+import io.dockstore.client.cli.nested.SingularityTest;
+import io.dockstore.common.CommonTestUtilities;
+import io.dockstore.common.SourceControl;
+import io.dropwizard.testing.ResourceHelpers;
+import io.swagger.client.ApiClient;
+import io.swagger.client.api.WorkflowsApi;
+import io.swagger.client.model.Workflow;
+import org.apache.commons.io.FileUtils;
+import org.bouncycastle.util.io.TeeOutputStream;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
+
+@Category(SingularityTest.class)
+public class SingularityIT extends BaseIT {
+
+    @ClassRule
+    public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+
+    @Before
+    @Override
+    public void resetDBBetweenTests() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false);
+    }
+
+    /**
+     * Tests that a simple CWL workflow can be run in Singularity instead of Docker
+     */
+    @Test
+    public void runCwlWorkflow() {
+        // manually register the CWL version of the md5sum-checker workflow locally
+        testingPostgres.runUpdateStatement("update enduser set isadmin = 't' where username = 'DockstoreTestUser2';");
+        final ApiClient webClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        Workflow workflow = workflowApi.manualRegister(SourceControl.GITHUB.getFriendlyName(), "DockstoreTestUser2/md5sum-checker",
+                "/checker-workflow-wrapping-workflow.cwl", "test", "cwl", null);
+        workflowApi.refresh(workflow.getId());
+
+        // start saving the output so as to make sure that Singularity actually runs instead of Docker
+        PrintStream old = System.out;  // save the regular output stream so it can be reset later
```

Comment:
We use a System Rule for this kind of thing, there should be examples that look somewhat like this https://stefanbirkner.github.io/system-rules/#SystemErrAndOutRule